### PR TITLE
gh10357 Add warehouseCode in OLCand

### PR DESCRIPTION
**Pull request by adi-stefan**
https://github.com/metasfresh/metasfresh/issues/10357

### CODE REVIEW

File: backend/de.metas.adempiere.adempiere/base/src/main/java-legacy/org/compiere/model/PO.java [hunk 1]
```diff
@@ -740,7 +740,7 @@ public final String get_ValueAsString(final String variableName)
 		final Object value = get_Value(variableName);
 		if (value == null)
 		{
-			return "";
+			return null;
```
Comment (metas-ts) [inline backend/de.metas.adempiere.adempiere/base/src/main/java-legacy/org/compiere/model/PO.java:743]: pls update the javadoc

File: backend/de.metas.business.rest-api/src/main/java/de/metas/rest_api/ordercandidates/request/JsonOLCandCreateRequest.java [hunk 2]
```diff
@@ -177,6 +177,10 @@
 			value = "External reference (document number) on a remote system. Not neccesarily unique, but but the external user will want to filter recrods using it")
 	String poReference;
 
+	@ApiModelProperty(required = false, //
+			value = "Translates to `C_OLCand.M_Warehouse_ID`.")
```
Comment (metas-ts) [inline backend/de.metas.business.rest-api/src/main/java/de/metas/rest_api/ordercandidates/request/JsonOLCandCreateRequest.java:181]:
```suggestion
			value = "Translates to `M_Warehouse.Value`. The looked up warehouse's ID is then set to `C_OLCand.M_Warehouse_ID`.")
```

File: backend/de.metas.salescandidate.base/src/main/java/de/metas/ordercandidate/api/OLCandOrderFactory.java [hunk 3]
```diff
@@ -155,7 +155,8 @@ private I_C_Order newOrder(final OLCand candidateOfGroup)
 		//
 		// use values from orderDefaults when the order candidate doesn't have such values
 		order.setC_DocTypeTarget_ID(DocTypeId.toRepoId(orderDefaults.getDocTypeTargetId()));
-		order.setM_Warehouse_ID(WarehouseId.toRepoId(orderDefaults.getWarehouseId()));
+		final WarehouseId warehouseId = candidateOfGroup.getWarehouseId();
+		order.setM_Warehouse_ID(WarehouseId.toRepoId(warehouseId == null ? orderDefaults.getWarehouseId() : warehouseId) );
```
Comment (metas-ts) [inline backend/de.metas.salescandidate.base/src/main/java/de/metas/ordercandidate/api/OLCandOrderFactory.java:159]: pls use `CoalesceUtils.coalesce(...)` instead

**Review by metas-ts — CHANGES_REQUESTED** [review]
nice! please check out my minor requests

File: backend/de.metas.adempiere.adempiere/base/src/main/java-legacy/org/compiere/model/PO.java (hunk 1 above)
Comment (adi-stefan) [reply backend/de.metas.adempiere.adempiere/base/src/main/java-legacy/org/compiere/model/PO.java:743]: Fixed javadoc.

**Review by metas-ts — APPROVED** [review]

**Pull request merged**